components/org.wso2.identity.webhook.wso2.event.handler/src/main/java/org/wso2/identity/webhook/wso2/event/handler/builder/WSO2LoginEventPayloadBuilder.java

```diff
@@ -0,0 +1,128 @@
+import org.wso2.identity.webhook.wso2.event.handler.model.WSO2AuthenticationSuccessEventPayload;
+import org.wso2.identity.webhook.wso2.event.handler.model.WSO2BaseEventPayload;
+import org.wso2.identity.webhook.wso2.event.handler.model.common.Application;
+import org.wso2.identity.webhook.wso2.event.handler.model.common.Organization;
+import org.wso2.identity.webhook.wso2.event.handler.model.common.User;
+import org.wso2.identity.webhook.wso2.event.handler.model.common.UserClaim;
+import org.wso2.identity.webhook.wso2.event.handler.model.common.UserStore;
+import org.wso2.carbon.identity.application.authentication.framework.context.AuthenticationContext;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.application.common.model.ClaimMapping;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.identity.data.publisher.authentication.analytics.login.model.AuthenticationData;
+import org.wso2.identity.event.publishers.common.model.EventPayload;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static org.wso2.identity.webhook.common.event.handler.constant.Constants.EVENT_SCHEMA_TYPE_WSO2;
+import static org.wso2.identity.webhook.common.event.handler.util.EventHookHandlerUtils.getReference;
+
+/**
+ * WSO2 Login Event Payload Builder.
+ */
+public class WSO2LoginEventPayloadBuilder implements LoginEventPayloadBuilder {
+
+    @Override
+    public EventPayload buildAuthenticationSuccessEvent(EventData eventData) {
+
+        WSO2AuthenticationSuccessEventPayload payload = new WSO2AuthenticationSuccessEventPayload();
+        buildEventPayload(eventData, payload);
+        return payload;
+    }
+
+    @Override
+    public EventPayload buildAuthenticationFailedEvent(EventData eventData) {
+
+        WSO2AuthenticationFailedEventPayload payload = new WSO2AuthenticationFailedEventPayload();
+        buildEventPayload(eventData, payload);
+        return payload;
+    }
+
+    @Override
+    public String getEventSchemaType() {
+
+        return EVENT_SCHEMA_TYPE_WSO2;
+    }
+
+    private void buildEventPayload(EventData eventData, WSO2BaseEventPayload payload) {
+
+        User user = new User();
+        AuthenticationData<?, ?> authenticationData = eventData.getAuthenticationData();
+        AuthenticatedUser authenticatedUser = eventData.getAuthenticatedUser();
+        AuthenticationContext authenticationContext = eventData.getAuthenticationContext();
+
+        populateUserAttributes(authenticatedUser, user);
+
+        user.setId(authenticationData.getUserId());
+        user.setRef(getReference(authenticationContext.getTenantDomain(),
+                Constants.SCIM2_ENDPOINT, authenticationData.getUserId()));
+        payload.setUser(user);
+        payload.setOrganization(new Organization(IdentityTenantUtil.getTenantId(authenticationContext.getTenantDomain()),
```

Comment:
Tenant id and org id is different as per my understanding. Let's recheck

Reply:
fixed with https://github.com/wso2-extensions/identity-webhook-event-handlers/pull/1/commits/f196ef27141312ec28d2fb649831a766aecadbf8